pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -603,9 +603,15 @@ public void segmentStoppedConsuming(LLCSegmentName llcSegmentName, String instan
     try {
       HelixHelper.updateIdealState(_helixManager, realtimeTableName, idealState -> {
         assert idealState != null;
-        idealState.getInstanceStateMap(segmentName).put(instanceName, RealtimeSegmentOnlineOfflineStateModel.OFFLINE);
+        Map<String, String> stateMap = idealState.getInstanceStateMap(segmentName);
+        if (stateMap != null) {
+          String state = stateMap.get(instanceName);
+          if (state != null && state.equals(RealtimeSegmentOnlineOfflineStateModel.CONSUMING)) {
+            stateMap.put(instanceName, RealtimeSegmentOnlineOfflineStateModel.OFFLINE);
+          }
```

Comment:
Please add some log such as cannot find instance in CONSUMING state for debugging purpose

Reply:
ok